docs/src/pages/playground.tsx

```diff
@@ -0,0 +1,72 @@
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
+import Layout from '@theme/Layout';
+
+import styles from './playground.module.css';
+import { useState } from 'react';
+import PlaygroundRenderSettings from '../components/PlaygroundRenderSettings';
+import PlaygroundPreview from '../components/PlaygroundPreview';
+import PlaygroundCodeEditor from '../components/PlaygroundCodeEditor';
+import toast, { Toaster } from 'react-hot-toast';
+import { renderImage } from '../api';
+
+function Homepage(): JSX.Element {
+  const [scene, setScene] = useState<object | Error>({
+    type: 'view',
+  });
+
+  const [responseData, setResponseData] = useState({
+    imageUrl: '',
+    errorMessage: '',
+  });
+
+  const setErrorMessage = message => {
+    setResponseData(prevResponseData => ({ ...prevResponseData, errorMessage: message }));
+  };
+
+  const handleCodeEditorChange = (scene: object | Error): void => {
+    setScene(scene);
+  };
+
+  const handleSubmit = async (): Promise<void> => {
+    try {
+      const blob = await renderImage(scene);
+      const imageObjectURL = URL.createObjectURL(blob);
+
+      setResponseData({ imageUrl: imageObjectURL, errorMessage: '' });
+    } catch (error: any) {
+      setErrorMessage(error.message);
+      toast.error(`${error.message}`);
+    }
+  };
+
+  return (
+    <div className={styles.page}>
+      <div className={styles.leftSide}>
+        <div className={styles.codeEditor}>
+          <PlaygroundCodeEditor
+            onChange={handleCodeEditorChange}
+            initialCodeEditorContent={JSON.stringify(scene, null, 2)}
```

Comment:
you are using state for this value, instead:
- move the initial value of scene to global const, e.g `const INITIAL_SCENE = { ...`
- create second global const e.g. `const INITIAL_SCENE_STRING = JSON.stringify(STARTING_SCENE, ...`